### PR TITLE
Add initial support for GCP and first GCP technique 

### LINKED ISSUE #53: Support GCP attack techniques

### PR DESCRIPTION
**Pull request by rileydakota**
### What does this PR do?

* New attack technique
* Enhancement

Adds initial support for GCP techniques, and initial GCP Technique 

closes #53 #128 

### Motivation

Adds initial provider for GCP, and initial GCP Attack Technique.

Notes on initial GCP Provider:
- I decided against abstracting away client creation, as by default with the google cloud SDK, Application Default Credential flow will be used (see https://cloud.google.com/docs/authentication/production#automatically and https://pkg.go.dev/cloud.google.com/go#hdr-Authentication_and_Authorization for more details). Happy to refactor this instead to a factory that abstracts it away if it better matches the other provider patterns in the codebase.
- I verify auth by checking for the presence of a GCP project variable (the GCP TF Provider documentation has all possible ones listed: https://registry.terraform.io/providers/hashicorp/google/latest/docs/guides/provider_reference#full-reference ), and by trying to instantiate a GCP SDK Client, which will raise an error if credentials aren't found
- Stratus Red Team user agent is injected
Screen Shot 2022-07-17 at 7 45 29 PM

Notes on initial technique:
- Seen in the wild: https://expel.com/blog/incident-report-spotting-an-attacker-in-gcp/
- Marked as `IsIdempotent`. There are no immediate consequences to firing without reverting, but a GCP Service Account can only have 10 keys, and didn't want to manage state of keys. 
- Currently writes the service account key payload to console so it could be used by the operator (wanted to follow the pattern in similar techniques)

### Checklist

- [x] The attack technique emulates a single attack step, not a full attack chain
- [x] We have factual evidence & references that the attack technique was used by real malware, pentesters, or attackers
- [x] The attack technique makes no assumption about the state of the environment prior to warming it up

### CODE REVIEW

**Comment by christophetd** [comment]
closes #53

File: internal/attacktechniques/gcp/persistence/create-service-account-key/main.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+)
+
+//go:embed main.tf
+var tf []byte
+
+func init() {
+	stratus.GetRegistry().RegisterAttackTechnique(&stratus.AttackTechnique{
+		ID:                         "gcp.persistence.create-service-account-key",
+		FriendlyName:               "Create a GCP Service Account Key",
+		Description:                ``,
```
Comment (christophetd) [inline internal/attacktechniques/gcp/persistence/create-service-account-key/main.go:24]: - [ ] Add a description

**Comment by christophetd** [comment]
Left a few comments, overall it looks great and we should definitely be able to merge it this week!

File: internal/attacktechniques/gcp/persistence/create-service-account-key/main.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+		Platform:                   stratus.GCP,
+		IsIdempotent:               false,
```
[the remark this answers is not in the document]
Comment (rileydakota) [reply internal/attacktechniques/gcp/persistence/create-service-account-key/main.go:44]: So this is why I shouldn't write PRs when tired 😂 - non-idempotent is correct and what I meant, apologies for the confusion.

File: internal/attacktechniques/gcp/persistence/create-service-account-key/main.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+		PrerequisitesTerraformCode: tf,
+		Detonate:                   detonate,
+		Revert:                     revert,
+	})
+}
+
+func detonate(params map[string]string) error {
+	saEmail := params["sa_email"]
+	ctx := context.Background()
+	service, err := iam.NewService(ctx, providers.GCP().GetUserAgentOption())
```
[the remark this answers is not in the document]
Comment (rileydakota) [reply internal/attacktechniques/gcp/persistence/create-service-account-key/main.go]: Sadly I wasn't able to find anything regarding setting that option globally.  Refactored to `providers.GCP().Options()` as suggested. 

Alternatively - we could also look at a factory/decorator style pattern.

[the remark this answers is not in the document]
Comment (christophetd) [reply internal/attacktechniques/gcp/persistence/create-service-account-key/main.go]: looks good!

**Pull request merged**